program/src/processor.rs

```diff
@@ -0,0 +1,485 @@
+//! Program processor
+
+use {
+    crate::{
+        check_id,
+        instruction::ProgramInstruction,
+        state::{
+            AddressLookupTable, LookupTableStatus, ProgramState, LOOKUP_TABLE_MAX_ADDRESSES,
+            LOOKUP_TABLE_META_SIZE,
+        },
+    },
+    solana_program::{
+        account_info::{next_account_info, AccountInfo},
+        clock::{Clock, Slot},
+        entrypoint::ProgramResult,
+        msg,
+        program::{invoke, invoke_signed},
+        program_error::ProgramError,
+        pubkey::{Pubkey, PUBKEY_BYTES},
+        rent::Rent,
+        slot_hashes::MAX_ENTRIES,
+        system_instruction,
+        sysvar::Sysvar,
+    },
+};
+
+// [Core BPF]: Locally-implemented
+// `solana_sdk::program_utils::limited_deserialize`.
+fn limited_deserialize<T>(input: &[u8]) -> Result<T, ProgramError>
+where
+    T: serde::de::DeserializeOwned,
+{
+    solana_program::program_utils::limited_deserialize(
+        input, 1232, // [Core BPF]: See `solana_sdk::packet::PACKET_DATA_SIZE`
+    )
+    .map_err(|_| ProgramError::InvalidInstructionData)
+}
+
+// [Core BPF]: Feature "FKAcEvNgSY79RpqsPNUV5gDyumopH4cEHqUxyfm8b8Ap"
+// (relax_authority_signer_check_for_lookup_table_creation) is now enabled on
+// all clusters, so the relevant checks have not been included in the Core BPF
+// implementation.
+// - Testnet:       Epoch 586
+// - Devnet:        Epoch 591
+// - Mainnet-Beta:  epoch 577
+fn process_create_lookup_table(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    untrusted_recent_slot: Slot,
+    bump_seed: u8,
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+
+    let lookup_table_info = next_account_info(accounts_iter)?;
+    let authority_info = next_account_info(accounts_iter)?;
+    let payer_info = next_account_info(accounts_iter)?;
+    let _system_program_info = next_account_info(accounts_iter)?;
+
+    if !payer_info.is_signer {
+        msg!("Payer account must be a signer");
+        return Err(ProgramError::MissingRequiredSignature);
+    }
+
+    // [Core BPF]: Since the `SlotHashes` sysvar is not available to BPF
+    // programs, checking if a slot is a valid recent slot must be done
+    // differently.
+    // The `SlotHashes` sysvar stores up to `512` recent slots (`MAX_ENTRIES`).
+    // We can instead use the `Clock` sysvar and do this math manually.
+    //
+    // Note this will no longer consider skipped slots wherein a block was not
+    // produced.
+    let derivation_slot = {
+        let clock = <Clock as Sysvar>::get()?;
+        let oldest_possible_slot = clock.slot.saturating_sub(MAX_ENTRIES as u64);
+        if untrusted_recent_slot > oldest_possible_slot && untrusted_recent_slot <= clock.slot {
```

Comment:
Nit to be consistent with the previous version
```suggestion
        if untrusted_recent_slot >= oldest_possible_slot && untrusted_recent_slot < clock.slot {
```